File: converters/TET-3.0-Linux/bind/perl/tetml.pl
#!/usr/bin/perl
# TET sample application for dumping PDF information in the XML language TETML
#
# $Id: tetml.pl,v 1.4 2008/12/23 17:31:11 rjs Exp $
#/

use tetlib_pl 3.0;
use strict;

# global option list */
my $globaloptlist = "searchpath={../data ../../../resource/cmap}";

# document-specific option list */
my $basedocoptlist = "";

# page-specific option list */
my $pageoptlist = "granularity=word";

# set this to 1 to generate TETML output in memory */
my $inmemory = 0;

eval  {
    my $tet;
    my $tetml;
    my $pageno;
    my $n_pages;
    my $docoptlist;
    my $doc;
    my $len;

    if ($#ARGV !=1) {
	die("usage: tetml <pdffilename> <xmlfilename>\n");
    }

    $tet = TET_new();

    TET_set_option($tet, $globaloptlist);

    if ($inmemory) {
	$docoptlist = sprintf("tetml={} %s", $basedocoptlist);
    } else {
	$docoptlist = sprintf("tetml={filename={%s}} %s",
	    $ARGV[1], $basedocoptlist);
    }

    $doc = TET_open_document($tet, $ARGV[0], $docoptlist);

    if ($doc == -1) {
	die(sprintf("Error %d in %s(): %s\n",
	    TET_get_errnum($tet), TET_get_apiname($tet), TET_get_errmsg($tet)));
    }

    $n_pages = TET_pcos_get_number($tet, $doc, "length:pages");

    # loop over pages in the document */
    for ($pageno = 1; $pageno <= $n_pages; ++$pageno) {
	TET_process_page($tet, $doc, $pageno, $pageoptlist);
    }

    # This could be combined with the last page-related call */
    TET_process_page($tet, $doc, 0, "tetml={trailer}");

    if ($inmemory) {
	my $fname = $ARGV[1];
	open(FP, ">$fname") || die(
	    sprintf("tetml: couldn't open output file '%s'\n", $fname));

	# Retrieve the generated TETML data from memory. Since we have
	# only a single call the result will contain the full TETML.
	#/

	$tetml = TET_get_xml_data($tet, $doc, "");
	if (!$tetml) {
	    die("tetml: couldn't retrieve XML data\n");
	}

	print FP $tetml;
	close(FP);
    }

    TET_close_document($tet, $doc);
};

if ($@) {
    printf("Error $@\n");
    exit;
}

File: converters/TET-3.0-Linux/bind/python/tetml.py
# TET sample application for dumping PDF information in the XML language TETML
#
# $Id: tetml.py,v 1.4 2008/12/23 17:31:12 rjs Exp $
#/

import sys
from tetlib_py import *

# global option list */
globaloptlist = "searchpath={../data ../../../resource/cmap}"

# document-specific option list */
basedocoptlist = ""

# page-specific option list */
pageoptlist = "granularity=word"

# set this to 1 to generate TETML output in memory */
inmemory = 0


if len(sys.argv) != 3:
    raise Exception("usage: tetml <pdffilename> <xmlfilename>\n")

try:
    try:
	pageno = 0
	tet = TET_new()

        TET_set_option(tet, globaloptlist)

	if (inmemory):
	    docoptlist =  "tetml={} %s" % basedocoptlist
	else:
	    docoptlist =  "tetml={filename={%s}} %s" % \
				(sys.argv[2], basedocoptlist)

        doc = TET_open_document(tet, sys.argv[1], docoptlist)

        if (doc == -1):
            raise Exception("Error %d in %s(): %s" % \
               (TET_get_errnum(tet), TET_get_apiname(tet), TET_get_errmsg(tet)))

        n_pages = int(TET_pcos_get_number(tet, doc, "length:pages"))

	# loop over pages in the document */
	for pageno in range(1, n_pages+1):
            TET_process_page(tet, doc, pageno, pageoptlist)

	# This could be combined with the last page-related call */
	TET_process_page(tet, doc, 0, "tetml={trailer}")

	if (inmemory):
            fp = open(sys.argv[2], "wb")

	    # Retrieve the generated TETML data from memory. Since we have
	    # only a single call the result will contain the full TETML.
	    #/

	    tetml = TET_get_xml_data(tet, doc, "")
	    if (not tetml):
		raise Exception("tetml: couldn't retrieve XML data")

	    fp.fwrite(tetml)
	    fp.close()

        TET_close_document(tet, doc)

    except TETlibException:
	print("\nError " + repr(TET_get_errnum(tet)) + "in " + \
		TET_get_apiname(tet) + "() on page " + repr(pageno) \
		+ ": " + TET_get_errmsg(tet) + "\n")

    except Exception, e:
	print "Exception occurred:"
	print e

finally:
    TET_delete(tet)
